src/main/java/com/chrosciu/bootcamp/tasks/github/GithubClient.java

```diff
@@ -3,29 +3,40 @@
 import com.chrosciu.bootcamp.tasks.github.dto.Branch;
 import com.chrosciu.bootcamp.tasks.github.dto.Repository;
 import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
 import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import retrofit2.HttpException;
 
+import java.time.Duration;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Slf4j
 @RequiredArgsConstructor
 public class GithubClient {
     private final GithubApi githubApi;
 
     public Flux<Repository> getUserRepositories(String username) {
-        //TODO: Implement
-        return null;
+        Mono<List<Repository>> userRepositories = githubApi.getUserRepositories(username);
+        return userRepositories.flatMapMany(Flux::fromIterable);
     }
 
     public Flux<Branch> getUserRepositoryBranches(String username, String repo) {
-        //TODO: Implement
-        return null;
+        Mono<List<Branch>> userRepositoryBranches = githubApi.getUserRepositoryBranches(username, repo);
+        return userRepositoryBranches.flatMapMany(Flux::fromIterable);
     }
 
     public Flux<Repository> getUsersRepositories(Flux<String> usernames) {
-        //TODO: Implement
-        return null;
+        return usernames.flatMap(username -> getUserRepositories(username)
+                .doOnError(e -> log.info(e.toString()))
+                .onErrorResume(e -> Mono.empty()));
```

Comment:
To nie zadziała do końca dobrze - owszem nie spowoduje wyrzucenia błędu, ale jako tako i tak przerwie pobieranie repozytoriów dla kolejny użytkowników po tym błędnym.